src/v/cluster/topic_recovery_status_types.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "serde/envelope.h"
+
+#include <compare>
+#include <type_traits>
+
+namespace cluster {
+
+struct status_request
```

Comment:
`operator == ()` should be defined explicitly

Reply:
added in https://github.com/redpanda-data/redpanda/pull/8567